app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumber.kt

```diff
@@ -129,6 +130,12 @@ class AddHousenumber : OsmElementQuestType<HousenumberAnswer> {
     override fun isApplicableTo(element: Element): Boolean? =
         if (!buildingsWithMissingAddressFilter.matches(element)) false else null
 
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("""nodes, ways, relations with
+            (addr:housenumber or addr:housename)
+            and !amenity and !craft and !leisure and !office and !shop and !tourism
```

Comment:
StreetComplete only asks housenumbers for buildings, so maybe instead do something like this?
```kotlin
.filter { it.tags.keys.all { key -> key.startsWith("addr:") || key == "building" } }
```
i.e. only show addresses that are "pure" addresses or that are buildings.

It may be more future proof

Reply:
Hmm, then I start thinking about all of the other valid things that could be on an address node: `entrance`, `source:*` and maybe others I haven't thought of...

Reply:
True